osu.Game/Skinning/SkinImporter.cs

```diff
@@ -43,6 +46,35 @@ public SkinImporter(Storage storage, RealmAccess realm, IStorageResourceProvider
 
         private const string unknown_creator_string = @"Unknown";
 
+        /// <summary>
+        /// Update an existing skin with the contents of a path
+        /// </summary>
+        /// <param name="notification">The progress notification</param>
+        /// <param name="task">The <see cref="ImportTask"/> to update the <paramref name="original"/> with</param>
+        /// <param name="original">The <see cref="SkinInfo"/> to update</param>
+        /// <returns></returns>
+        public override Task<Live<SkinInfo>?> ImportAsUpdate(ProgressNotification notification, ImportTask task, SkinInfo original)
+        {
+            var skinInfoLive = original.ToLive(Realm);
+
+            skinInfoLive.PerformWrite(skinInfo =>
+            {
+                // Not sure if this deletes the files from the storage or just the database.
+                skinInfo.Files.Clear();
```

Comment:
As the comment states. Does this also flag the files for deletion? I tried to test it but couldn't conclude whether it does or doesn't. And `ModelManager` seems to do roughly the same in its [`DeleteFile`](https://github.com/ppy/osu/blob/b7883f18be12d304fddb57269c5bcb123e0763b2/osu.Game/Database/ModelManager.cs#L70) method. So I'm a bit confused here.

Reply:
Don't add comments for this. Ask elsewhere.

And yes, it does. Storage is managed as a separate layer based on file references.